### PR TITLE
feat(#12): add otel service name as an input or env var 

### LINKED ISSUE #12: Not able to overwrite service name with OTEL_SERVICE_NAME
Even though I'm setting:
```
env:
      OTEL_SERVICE_NAME: github-actions-metadata
````

I'm not able to overwrite the OTEL service name, and the only option is to get the workflow name.

### PR DESCRIPTION
**Pull request by pragmaticivan**
fixes #12 

ref: https://github.com/open-telemetry/opentelemetry-specification/blob/f124b21de091f3d668e4ce11b954ec17e3da231f/specification/sdk-environment-variables.md#general-sdk-configuration

### CODE REVIEW

**Review by v1v — DISMISSED** [review]

File: src/tracing/trace.ts [hunk 1]
```diff
@@ -24,9 +24,11 @@ function stringToHeader(value: string): StringDict {
 export function createTracerProvider(
   otlpEndpoint: string,
   otlpHeaders: string,
+  otelServiceName: string,
   workflowRunJobs: WorkflowRunJobs
```
Comment (nikordaris) [inline src/tracing/trace.ts:28]:
```suggestion
export function createTracerProvider(
  otlpEndpoint: string,
  otlpHeaders: string,
  workflowRunJobs: WorkflowRunJobs,
  otelServiceName?: string | null | undefined,
```
Lets make this a non-breaking change to the API.

File: src/tracing/trace.test.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ describe("createTracerProvider", () => {
       subject = createTracerProvider(
         "localhost",
         "test=foo",
+        "",
```
Comment (nikordaris) [inline src/tracing/trace.test.ts:35]: We can remove all the empty strings in the tests after making the change non-breaking.

**Review by nikordaris — CHANGES_REQUESTED** [review]
We need to add a test case that validates the service name. You should be able to add the the test in `trace.test.ts` under the "resource attributes" test suite for a test case of "has service.name resource as otelServiceName" or something like that.

**Comment by nikordaris** [comment]
Thanks for the contribution @pragmaticivan! I left a couple of comments. After those changes i think this will be good to merge in.

**Comment by pragmaticivan** [comment]
Updated @nikordaris thanks!

**Review by nikordaris — DISMISSED** [review]

**Comment by pragmaticivan** [comment]
@nikordaris anything I need to fix? I just did run lint and test locally and they are all working.

**Comment by pragmaticivan** [comment]
I've just updated my node version to exactly the one used in the CI (which uses latest and will probably come with unexpected side effects).

By changing the `minor` version I was able to replicate it locally. Some packages are due with updates, so that might also be a good time to get it in.

**Comment by nikordaris** [comment]
@pragmaticivan ok, i think i fixed the build issues. if you pull the latest from main it should fix the build for this PR. apparently node 16.16 is stricter on version conflicts than 16.13

**Comment by ivan-santos-eq** [comment]
@nikordaris rebased from the `main` branch.

**Review by nikordaris — DISMISSED** [review]

**Comment by nikordaris** [comment]
Ok the update dist action wasn't designed for forks. I need to fix that.

**Comment by nikordaris** [comment]
@pragmaticivan ok i think i resolved the build issues. it looks like it created conflicts for you though. if you pull the latest, hopefully that will fix the build

**Comment by pragmaticivan** [comment]
@nikordaris rebased

**Pull request merged**